internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_controller.go

```diff
@@ -243,6 +269,16 @@ func (c *Controller) Unlock() bool {
 }
 
 func (c *Controller) run(ctx context.Context, seq runtime.Sequence, phases []runtime.Phase, data interface{}) error {
+	c.ctxMutex.Lock()
+	c.runningSequence = seq
```

Comment:
is there any reason we set `runningSequence` here, while other pieces are set in `Run()` ?

as we check for `runningSequence` in `Run()`, does it make sense to set the `runningSequence` there as well? and we should set it somehow atomically with a decision to run the sequence